src/hal/src/pso/mod.rs

```diff
@@ -206,6 +206,7 @@ pub struct Specialization {
 }
 
 /// Scalar specialization constant with value for overriding.
+/// DOC TODO, don't know what this means.
```

Comment:
specialization constants allow for easy configuration of multiple similar pipelines, e.g. you may have a boolean exposed to the shader that switches the specularity on/off, and provide it as a specialization constant. That would make on/off to be separate PSOs but they share most of the internal stuff and are fast to produce. More importantly, they are fast to execute, since the driver can optimize out the branch on that other PSO creation.